resources/packs/gcp/pubsub.go

```diff
@@ -0,0 +1,347 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/resources"
+	"go.mondoo.com/cnquery/resources/packs/core"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+func (g *mqlGcpProjectPubsub) id() (string, error) {
+	return "gcp.project.pubsub", nil
+}
+
+func (g *mqlGcpProjectPubsub) init(args *resources.Args) (*resources.Args, GcpProjectPubsub, error) {
+	if len(*args) > 0 {
+		return args, nil, nil
+	}
+
+	provider, err := gcpProvider(g.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	projectId := provider.ResourceID()
+	(*args)["projectId"] = projectId
+
+	return args, nil, nil
+}
+
+func (g *mqlGcpProject) GetPubsub() (interface{}, error) {
+	projectId, err := g.Id()
+	if err != nil {
+		return nil, err
+	}
+
+	return g.MotorRuntime.CreateResource("gcp.project.pubsub",
+		"projectId", projectId,
+	)
+}
+
+func (g *mqlGcpProjectPubsubTopic) id() (string, error) {
+	return g.Id()
```

Comment:
The id implementation do not need to have an id field. You can generate the id from any field.